tests/ZichtTest/Bundle/UrlBundle/Aliasing/ListenerTest.php

```diff
@@ -187,6 +187,41 @@ public function testRedirectModes($statusCode, $expectsException = false)
             $this->assertEquals('/bar', $setResponseValue->headers->get('location'));
         }
     }
+
+    /**
+     * @dataProvider statusModes
+     */
+    public function testRedirectSupportsQueryString($statusCode, $expectsException = false)
+    {
+        $event = $this->getMockBuilder('Symfony\Component\HttpKernel\Event\GetResponseEvent')
+            ->disableOriginalConstructor()
+            ->getMock()
+        ;
+        $event->expects($this->any())->method('getRequestType')->will($this->returnValue(HttpKernelInterface::MASTER_REQUEST));
+        $event->expects($this->atLeastOnce())->method('getRequest')->will($this->returnValue(\Symfony\Component\HttpFoundation\Request::create('/foo?is_there_more=yes')));
+        $setResponseValue = null;
+        $this->aliasing->expects($this->once())->method('hasInternalAlias')->will($this->returnValue(
+            new UrlAlias('/foo', '/bar', $statusCode)
+        ));
+        if ($expectsException) {
```

Comment:
ik snap heel de if else constructie niet. (bad smell) en je zou als je een exception wilt testen toch liever de bestaande PHPunit dingen kunnen gebruiken https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions